### ISSUE TITLE
Potential features

### ISSUE THREAD

**Issue by marlycormar** [pharmaR/riskassessment#176]
## Potential App Features
App Walk through:

#### General
- [x]  Make it supers obvious on the readme how to run the app for the first time (wait till app is a package) @marlycormar 
- [x] #198 (PR #199) - @marlycormar.
- [x] Selecting packages like "stanly" and "samplesizeCMH" cause the app to blow up.
We can either dis-allow these packages that don't have CRAN history, or fix the error and say
"Community Usage Metrics not available for this package" @Robert-Krajcik 
- [x] Click on MaintMetrics, select package, de-select package results in
`error: resulted in Error: Expecting a single string value: [type=character; extent=0]. @Jeff-Thompson12 
- [x] Report Preview into .docx format needs work @marlycormar @Robert-Krajcik 
- [x] Database overview -- download report fails (server problem) @AARON-CLARK 
- [x] Unable to see "admins_view" even though my role is admin  @Robert-Krajcik @marlycormar 
Not sure this is correct: req(input$res_auth$admin == TRUE)  -- This value is NULL 
- [x] The "CURRENT RISK SCORE WEIGHTS BY METRIC" is blank @Robert-Krajcik @marlycormar 

#### Uploaded packages 
- [x] When uploading a CSV, waitress loading bar for packages is not very informative about what package is getting accessed and if you upload a lot of packages your just left looking at percentages for a long time. Switch back to  @Robert-Krajcik 
- [x] #200 (PR #201) @marlycormar.
- [x] Table: Source field isn't self explanatory… data sourced from CRAN or Local? **Removing `source` for now** (PR #201)
- [x] Introjs not working if you upload a csv? @AARON-CLARK 

#### Sidebar
- [x] If you select a package and then select '-', you'll get an error: Expecting a single string value: [type=character; extent=0] (PR #203) @Jeff-Thompson12 
- [x] No visual feedback when you submit an initial overall comment and it doesn't update on the report preview tab when you submit a second overall comment  @Jeff-Thompson12 (PR #209)

#### Report preview
- [x] Download reports doesn't work there @AARON-CLARK 
- [x] Community usage metrics is missing @marlycormar 

#### Community Usage 
- [x] Make sure title and buttons don't overlap on the plot (fixed with #180).
- [x] Have a green color to display vertical lines?
- [x] Match the font? @marlycormar to send @AARON-CLARK the font name. (fixed with #180)

#### Database Overview
- [x] Not reactive… so it won't show a package you just uploaded (PR #201) @marlycormar 
- [x] Download reports not working @AARON-CLARK 
- [x] No admin table rendering! @Robert-Krajcik @marlycormar 

#### Administrator Mode (@Jeff-Thompson12 trying to look into this)
- [x] Log out button get's shoved to the right (PR #204)
- [x] Add User button is hard to identify/ find because it's so big, spanning the whole top of screen (PR #204)
- [x] May be hard to change: if you edit a user (like taking away admin privileges), you have to hit submit and THEN cancel because the modal won't close. (PR #204)
- [x] Create a single tab that has a credential manager AND the metric re-weighting UI and package database download features (which were both formerly on the database overview screen)

#### Other Features & Backlog
- [x] Footer is no longer on the page.
- [ ] Make assessment asynchronous so that the user can see metrics on some of the packages while the rest loads (reopen #5)
- [x] Modularize app (reopen #16). Remove code duplication (reopen #53).
- [ ] Create a report for each dependency (reopen #26)
- [ ] Automate adding custom metrics (reopen #27)
- [x] Document how to use SQLite (reopen #40)
- [ ] Add code coverage tab with covr_coverage metric once it is implemented in riskmetric (reopen #45)
- [x] Make app more portable (reopen #48).
- [x] Make the app a package (reopen #155).
- [x] Loading a non-existent package yields many un-helpful warning messages (reopen #49)
- [x] Documentation explorer: allows to explore the package help files (suggested by Andy).
- [x] Test explorer:  allows to explore/browse the test files. The user can select a function and the app will highlight the location of the test files for that function (suggested by Andy).
- [x] Display a list of reverse imports/dependencies (suggested by Andy).
- [ ] Update metrics dynamically with riskmetric (reopen #105 and see #179).
- [ ] Explain app structure from the point of view of a developer (reopen #58).

**Comment by Robert-Krajcik** [comment]
PR #177 also addresses issue #49

**Comment by Robert-Krajcik** [comment]
I'm working on the SQLite vignette.

**Comment by Jeff-Thompson12** [comment]
@marlycormar When you say "create a single tab that has a credential manager AND the metric re-weighting UI and package database download features (which were both formerly on the database overview screen)", is the desire for the metric reweighting to only be viewable by admin users?

**Comment by marlycormar** [comment]
@Jeff-Thompson12 that's exactly right. Admins are the only ones who can modify the weights of the metrics. So, it is a good idea to have a dedicated tab for admins (shinymanager UI + metric re-weighting UI)

**Comment by marlycormar** [comment]
From @Robert-Krajcik 

Hi @marlycormar and @AARON-CLARK and @Jeff-Thompson12 
Here are some outstanding issues I have found:

- 	IntroJs seems to be **mostly** working now -- I have added comments below.  
- 	Still using waitress function.
-  	Selecting packages like "stanly" and "samplesizeCMH" cause the app to blow up.
We can either dis-allow these packages that don't have CRAN history, or fix the error and say
"Community Usage Metrics not available for this package"

- 	Click on MaintMetrics, select package, de-select package results in
`error:
resulted in
Error: Expecting a single string value: [type=character; extent=0].`

- 	Report Preview into .docx format needs work.
- 	Database overview -- download report fails (server problem)
- 	Unable to see "admins_view" even though my role is admin
Not sure this is correct: req(input$res_auth$admin == TRUE)  -- This value is NULL

- 	The "CURRENT RISK SCORE WEIGHTS BY METRIC" is blank
